### PR TITLE
Implement beat-synchronized animation in skip overlay

### PR DESCRIPTION
**Pull request by tristaneku**
I think the new beat-synced animation in the break overlay is really cool, so I implemented pretty much the same approach in the skip overlay.
I also rounded the corners by replacing the box with a circle.

Let me know what you think.

https://github.com/user-attachments/assets/6bf3fe3f-421c-4b78-a52e-9875851414ac

### CODE REVIEW

**Comment by nekupaw** [comment]
ahh, I didn't mean to merge it directly. It's my first pull request, so sorry.

**Comment by nekupaw** [comment]
Yeah, I was surprised too. It happened after I ran a code formatter over it, or rather, entered the command dotnet format into the command line.
I'll fix it right away, just a moment please.

**Comment by bdach** [comment]
Yeah we don't use `dotnet format` anymore here.

**Comment by peppy** [comment]
Should not be merged without squash.

Also @nekupaw please stop merging master.

**Comment by nekupaw** [comment]
Sorry, I forgot to create a branch at the start. I just intended to include the new merges.

File: osu.Game/Screens/Play/SkipOverlay.cs [hunk 1]
```diff
@@ -210,6 +212,17 @@ public void OnReleased(KeyBindingReleaseEvent<GlobalAction> e)
...
         }
 
+        protected override void OnNewBeat(int beatIndex, TimingControlPoint timingPoint, EffectControlPoint effectPoint, ChannelAmplitudes amplitudes)
+        {
+            base.OnNewBeat(beatIndex, timingPoint, effectPoint, amplitudes);
+
+            if (fadeOutBeginTime <= gameplayClock.CurrentTime)
+                return;
+
+            float progress = (float)(gameplayClock.CurrentTime - displayTime) / (float)(fadeOutBeginTime - displayTime);
```
Comment (bdach) [inline osu.Game/Screens/Play/SkipOverlay.cs:222]: Why is `fadeOutBeginTime` checked against current time (therefore clamping from the left), but progress is not forcefully clamped to be no larger than 1?

**Review by bdach — CHANGES_REQUESTED** [review]
while addressing this review please fix history as requested by force-pushing (this PR should consist of a single commit)

**Comment by nekupaw** [comment]
I'll create a pull request using a new branch now, I think this is the easiest way.

**Pull request closed without merging**